File: knowmind-server/app/schemas/evaluation.py
from pydantic import BaseModel, Field


class EvalKpiOut(BaseModel):
    value: float
    delta: float


class EvalTrendPointOut(BaseModel):
    label: str
    faithfulness: float
    answer_relevancy: float
    context_recall: float
    context_precision: float


class EvalVersionCompareOut(BaseModel):
    name: str
    current: float
    baseline: float


class EvalStatsOut(BaseModel):
    total_runs: int
    question_count: int
    avg_latency_s: float
    pass_rate: float


class EvalDashboardOut(BaseModel):
    run_id: str | None = None
    created_at: str | None = None
    version: str | None = None
    mode: str = "stub"
    sample_count: int = 0
    kpis: dict[str, EvalKpiOut] = Field(default_factory=dict)
    trend: list[EvalTrendPointOut] = Field(default_factory=list)
    version_compare: list[EvalVersionCompareOut] = Field(default_factory=list)
    stats: EvalStatsOut = Field(
        default_factory=lambda: EvalStatsOut(
            total_runs=0,
            question_count=0,
            avg_latency_s=0.0,
            pass_rate=0.0,
        )
    )

File: knowmind-web/src/services/evaluation.ts
/**
 * RAG 评估看板 API（`/api/v1/evaluation/dashboard`）。
 */

import { apiFetch, parseApiError } from "@/services/http";

export type EvalKpiDto = {
  value: number;
  delta: number;
};

export type EvalTrendPointDto = {
  label: string;
  faithfulness: number;
  answer_relevancy: number;
  context_recall: number;
  context_precision: number;
};

export type EvalVersionCompareDto = {
  name: string;
  current: number;
  baseline: number;
};

export type EvalDashboardDto = {
  run_id: string | null;
  created_at: string | null;
  version: string | null;
  mode: string;
  sample_count: number;
  kpis: Record<string, EvalKpiDto>;
  trend: EvalTrendPointDto[];
  version_compare: EvalVersionCompareDto[];
  stats: {
    total_runs: number;
    question_count: number;
    avg_latency_s: number;
    pass_rate: number;
  };
};

export async function fetchEvalDashboard(): Promise<EvalDashboardDto> {
  const res = await apiFetch(`/evaluation/dashboard`, { });
  if (!res.ok) throw new Error(await parseApiError(res));
  return (await res.json()) as EvalDashboardDto;
}

export async function runEvalPipeline(): Promise<EvalDashboardDto> {
  const res = await apiFetch(`/evaluation/run`, {
    method: "POST",
    });
  if (!res.ok) throw new Error(await parseApiError(res));
  return (await res.json()) as EvalDashboardDto;
}
